core/drivers/crypto/caam/mp/caam_mp.c

```diff
@@ -0,0 +1,404 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2021, 2023 NXP
+ */
+#include <caam_hal_ctrl.h>
+#include <caam_jr.h>
+#include <caam_mp.h>
+#include <caam_utils_mem.h>
+#include <caam_utils_status.h>
+#include <drivers/caam_extension.h>
+#include <kernel/pm.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+#include <tee/cache.h>
+#include <utee_defines.h>
+
+#define MP_SIGN_MAX_MSG_SIZE (4 * 1024)
+
+#ifdef CFG_PHYS_64BIT
+#define MP_PRIV_DESC_ENTRIES 7
+#define MP_PUB_DESC_ENTRIES 7
+#else
+#define MP_PRIV_DESC_ENTRIES 6
+#define MP_PUB_DESC_ENTRIES 6
+#endif
+
+/*
+ * Manufacturing passphrase and MPMR values.
+ */
+static const char g_passphrase[] = "manufacturing protection";
```

Comment:
this is a private passphrase? or a public identifier?

Reply:
Private. This passphrase and a secret embedded into the silicon is used to generate the MP key pair.

Reply:
Why the `g_` prefix?

By the way, minor detail, these two could be moved into `caam_mp_init()`.

Reply:
`g_` was for global. But you're right, this value can be moved to `caam_mp_init()`